### PR TITLE
WIP savq doc codereview

### PR DESCRIPTION
**Pull request by rktjmp**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -43,562 +49,131 @@ There are two interactive tutorials provided,
...
-can be chained.
-
-There are 3 main operations you may want to perform:
-
-- Rotate a hue.
-- Saturate (or desaturate) a color.
-- Lighten (or darken) a color.
+1. Define your colors using the HSL module
+2. Define your highlight groups with a Lush spec
+3. Organise your color scheme for normal usage and/or distribution
```
Comment (rktjmp) [inline README.md:60]: @savq

I understand the order here, not getting stuck in file/folder creation before getting to play around, but I think not getting the correct setup at the start leads to bad inertia later on.

There is the lush-template folder in the examples folder, maybe step zero might be copying that folder somewhere and doing some basic renames? A problem with this is that people need to find the folder which may be unclear when you're using a plugin manager.

I did consider, and maybe regret not doing it now,  creating the template in its own repo, which would give "lush theme index" "for free" via the gh forks list. Maybe I will revisit that idea.

File: README.md [hunk 2]
```diff
@@ -43,562 +49,131 @@ There are two interactive tutorials provided,
...
 
-- 2020-11-23
-  - Lush.ify now reports errors in a more consistent format.
-  - Lush.ify now rate-limits eval attempts on parsing errors.
-- 2020-11-21
-  - Lush-spec now supports group inheritance.
-- 2020-11-19
-  - Initial release.
+Lush also provides functions to export your color scheme in a more automated way.
+See the related documentation. <!-- link to parse-compile-apply -->
```
Comment (rktjmp) [inline README.md:179]: Yep, probably link out to the vim docs or API.md or something. I will handle that once the readme settles.

File: README.md [hunk 3]
```diff
@@ -43,562 +49,131 @@ There are two interactive tutorials provided,
...
-- `color.abs_saturate(n)` (and `abs_desaturate`)
-- `color.abs_lighten(n)` (and `abs_darken`)
-
-Behaves as:
-
-```lua
-color = hsl(0, 50, 50)
-color.abs_saturate(10) -- adds 10 to 50
-```
+Note that cyan is also the complement of red in RGB: `(255, 0, 0)' == (0, 255, 255)`,
```
Comment (rktjmp) [inline README.md:82]: Extra `'`

**Comment by savq** [comment]
Ok, I fixed some of the suggestions. I agree that explaining the directory structure first is better.

I also put back the additional info that doesn't really fit in the docs.

One of the things I noted that could still be improved is the usage of the words spec, theme and color scheme; or rather, we could remove the instances of _theme_, and replace it with spec, when talking about Lua, or color scheme, when talking about VimL (or the parsed result); or simply call it _colors_, like `lush_colors/` instead of `lush_theme`.

**Comment by rktjmp** [comment]
I've done a bit more work, https://github.com/rktjmp/lush.nvim/tree/readme-update (cross branch diff: https://github.com/savq/lush.nvim/compare/doc...rktjmp:readme-update).

I think I might compress 2 and 3 into one section, right now it reads like you have to define your colours *then* the spec, when really they can be intermixed, or *more interestingly* apply HSL operations on group attributes.

Really I think I might drop quite a lot of it and just have the demo spec for githubbers to see what they can expect to write and then point any actual guide to docs/lush.txt and :LushRunTutorial.

**Comment by savq** [comment]
> I think I might compress 2 and 3 into one section … apply HSL operations on group attributes

Yeah, re-reading that does make it seem like the two things are more separate than they should be.


> Really I think I might drop quite a lot of it ... then point any actual guide to docs/lush.txt and :LushRunTutorial.

Sure. I actually found it quite difficult to explain _why_ Lush is easier to use than other tools, even though it's obvious after running the tutorial. Maybe put a bigger demo video? something that shows code in a split buffer.

The rest looks pretty good to me already. I liked the idea of a script to copy the template automatically. I thought it could be added to the repo, with something to check for `LUSH_NAME`:
```sh
#!/bin/sh
[[ -z "${LUSH_NAME}" ]] && echo 'LUSH_NAME is not defined' && exit 1
# ... rest of the script
```

Other than that, I don't know if there's anything missing.

**Comment by savq** [comment]
I also saw that you put paq in there 👀 😄

**Comment by rktjmp** [comment]
Manually merged offline.

**Pull request closed without merging**